src/integration_test/java/com/klodnicki/Bike/v2/rest/controller/RentBikeControllerIntegrationTest.java

```diff
@@ -0,0 +1,226 @@
+                    assertNotNull(bikeResponseDTO);
+                    assertEquals(expected, bikeResponseDTO);
+                });
+    }
+
+    @Test
+    public void rentBike_ShouldInstantiateAndReturnRentResponseDTOAndSaveRentInDatabase_WhenRentRequestIsGiven() {
+
+        bike.setChargingStation(chargingStation);
+        bikeRepository.save(bike);
+        RentRequest rentRequest = new RentRequest(null, user.getId(), bike.getId(), 10);
+
+        webTestClient.post()
+                .uri("/api/rentals/add")
+                .bodyValue(rentRequest)
+                .exchange()
+                .expectStatus().isOk()
+                .expectBody(RentResponseDTO.class)
+                .consumeWith(response -> {
+                    RentResponseDTO responseDTO = response.getResponseBody();
+                    assertNotNull(responseDTO);
+                    assertTrue(rentRepository.count() > 0);
+                    assertEquals(rentRequest.getDaysOfRent(), responseDTO.getDaysOfRent());
+                    assertEquals(user.getName(), responseDTO.getUserForNormalUserResponseDTO().getName());
+                    assertEquals(user.getRole(), responseDTO.getUserForNormalUserResponseDTO().getRole());
+                    assertEquals(bike.getSerialNumber(), responseDTO.getBikeForNormalUserResponseDTO().getSerialNumber());
+                    assertEquals(bike.getBikeType(), responseDTO.getBikeForNormalUserResponseDTO().getBikeType());
+                });
+    }
+
+    @Test
+    public void updateRent_ShouldUpdateDaysOfRentAndMapAndReturnRentResponseDTO_WhenRentIdAndRentRequestDTOIsGiven() {
+
+        Rent rent = new Rent(null, LocalDateTime.of(2023, 11, 23, 10, 0, 0),
+                null, 10, 100.00, bike, user, null);
+        rentRepository.save(rent);
+
+        RentRequestDTO rentRequestDTO = new RentRequestDTO(null, LocalDateTime.of(2023, 11, 23,
+                10, 0, 0), null, 5, null,
+                null, null);
+
+        RentResponseDTO expected = new RentResponseDTO(1L, rentRequestDTO.getRentalStartTime(),
+                rentRequestDTO.getRentalEndTime(), rentRequestDTO.getDaysOfRent(), rentRequestDTO.getBikeForNormalUserResponseDTO(),
+                rentRequestDTO.getUserForNormalUserResponseDTO(), rentRequestDTO.getStationForNormalUserResponseDTO());
+
+        webTestClient.put()
+                .uri("/api/rentals/" + rent.getId())
+                .bodyValue(rentRequestDTO)
+                .exchange()
+                .expectStatus().isOk()
+                .expectBody(RentResponseDTO.class)
+                .consumeWith(response -> {
+                    RentResponseDTO responseDTO = response.getResponseBody();
+                    assertNotNull(responseDTO);
+                    assertEquals(expected.getDaysOfRent(), responseDTO.getDaysOfRent());
+                });
+
+        rent.setBike(null);
+        rent.setUser(null);
+        rentRepository.save(rent);
```

Comment:
What is the purpose of those lines here?

Reply:
If I dont add it then Im not able to delete these on db (they hold foreing keys and cascades dont work)